File: Programming Fundamentals/Labs/Objects and Classes/Songs.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Songs {
    static class Song {
        private String typeList;
        private String name;
        private String time;

        public Song (String typeList, String name, String time) {
            this.typeList = typeList;
            this.name = name;
            this.time = time;
        }

        public String getTypeList() {
            return typeList;
        }
        public String getName() {
            return name;
        }
        public String getTime() {
            return time;
        }
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int n = Integer.parseInt(scan.nextLine());

        List<Song> listSongs = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            String [] inputSongArr = scan.nextLine().split("_");
            Song currentSong = new Song(inputSongArr[0], inputSongArr[1], inputSongArr[2]);

            listSongs.add(currentSong);
        }

        String command = scan.nextLine();
        if (command.equals("all")) {
            for (Song item : listSongs) {
                System.out.println(item.getName());
            }
        }
        else {
            for (Song item : listSongs) {
                if (item.getTypeList().equals(command)) {
                    System.out.println(item.getName());
                }
            }
        }
    }
}

File: Front-End/Java Script/Labs/Objects and Classes/songs.js
function printSongsOfTypeList(inputData) {
    const n = inputData[0];
    const searchedTypeList = inputData[inputData.length - 1];

    class Song {
        constructor(typeList, name, time) {
            this.typeList = typeList;
            this.name = name;
            this.time = time;
        }

        getTypeList() {
            return this.typeList;
        }

        getName() {
            return this.name;
        }
    }

    for (let i = 1; i <= n; i++) {
        const [currentTypeList, currentName, currentTime] = inputData[i].split("_");

        const song = new Song(currentTypeList, currentName, currentTime);

        if (searchedTypeList === "all" || song.getTypeList() === searchedTypeList) {
            console.log(song.getName());
        }
    }
}

printSongsOfTypeList([2,
    'like_Replay_3:15',
    'ban_Photoshop_3:48',
    'all']
);